src/luagit2/libgit2/lua_libgit2.c

```diff
@@ -0,0 +1,259 @@
+#include "lua_libgit2.h"
+
+int lua_git_libgit2_init (lua_State *L) {
+	git_libgit2_init();
+	return 1;
+}
+
+int lua_git_libgit2_shutdown (lua_State *L) {
+	git_libgit2_shutdown();
+	return 1;
+}
+
+int lua_git_libgit2_features (lua_State *L) {
+	check_error_long(git_libgit2_features(),
+	    "Error Accessing libgit2 features", NULL);
+	int features_number = git_libgit2_features();
+	lua_pushinteger(L, features_number);
+	return 1;
+}
+
+int lua_git_libgit2_version (lua_State *L) {
+	libgit_version_data *version_data;
+
+	version_data = (libgit_version_data *)lua_newuserdata(L, sizeof(*version_data));
```

Comment:
Should we take a moment to think about the usability from the point of view of the person who is using the Lua library to run libgit2? Do you think it's worth to put this information in userdata? 

In my opinion, in idiomatic Lua, the best way to access this information while having a similar api to the original C library would be like this, returning 3 values: 
```
local major, minor, rev = this_function()
```

What do you think?

Let's take a moment to evaluate the implications of this and try to bring other functions also closer to a "Lua way of thinking" :)

Reply:
Yes It will definitively look much better.

Reply:
The integer on the last `return` is the number of results. So I think all you have to change it to 3 and use `lua_pushnumber(L, the_number)` for the versions, watch out for the fact this is a stack and it might affect the order where things are returned. Make sure to test it with different numbers to make sure it comes out correct :)